src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/IEventHubReceiver.cs

```diff
@@ -33,21 +34,61 @@ public interface IEventHubReceiver
     /// </summary>
     internal class EventHubReceiverProxy: IEventHubReceiver
     {
-        private PartitionReceiver receiver;
+        private readonly PartitionReceiver client;
 
-        public EventHubReceiverProxy(PartitionReceiver receiver)
-        {
-            this.receiver = receiver;
-        }
+        public EventHubReceiverProxy(EventHubPartitionSettings partitionSettings, string offset, ILogger logger)
+         {
+             var receiverOptions = new PartitionReceiverOptions();
+             if (partitionSettings.ReceiverOptions.PrefetchCount != null)
+             {
+                 receiverOptions.PrefetchCount = partitionSettings.ReceiverOptions.PrefetchCount.Value;
+             }
+
+             var options = partitionSettings.Hub;
+             this.client = options.TokenCredential != null
+                 ? new PartitionReceiver(options.ConsumerGroup, partitionSettings.Partition, GetEventPosition(), options.FullyQualifiedNamespace, options.Path, options.TokenCredential, receiverOptions)
+                 : new PartitionReceiver(options.ConsumerGroup, partitionSettings.Partition, GetEventPosition(), options.ConnectionString, options.Path, receiverOptions);
+
+             EventPosition GetEventPosition()
+             {
+                 EventPosition eventPosition;
+
+                 // If we have a starting offset, read from offset
+                 if (offset != EventHubConstants.StartOfStream)
+                 {
+                     if (!long.TryParse(offset, out var longOffset))
```

Comment:
I think we already check that in `EventHubCheckpointer.Load`, but it is still good do to it here. But I think we should log this as an error, and then treat as there was no checkpoint in storage (and respect the value of `ReceiverOptions.StartFromNow`)

Reply:
hummm make sense. I Just tried to not change the way it was implemented on master. I would rather prefer to add a PR later to enhance this and keep this one as close as possible to the PR we already have merged.